go/teams/hidden/manager.go

```diff
@@ -0,0 +1,257 @@
+package hidden
+
+import (
+	"fmt"
+	libkb "github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+	sig3 "github.com/keybase/client/go/sig3"
+	storage "github.com/keybase/client/go/teams/storage"
+)
+
+// ChainManager manages a hidden team chain, and wraps put/gets to mem/disk storage.
+// Accesses are single-flighted by TeamID. Implements that libkb.HiddenTeamChainManager
+// interface.
+type ChainManager struct {
```

Comment:
I'm not clear on the high level responsibilities of LoaderPackage and ChainManager.
During a team load, we make a LoaderPackage and in `Update`, check the well-formedness and the ratchets of new hidden data against what's on disk/in memory. If it checks out, the load continues and later we `Commit`, which calls `Chain.Manager.Advance` that checks ratchets again? What is the purpose of the ratchet check in `Update`?

Wrt where the ratchet advances, I only see it happening when we post a rotate. Should it be happening on Advance too?

Reply:
`ChainManager` is analogous to `FastTeamLoader` and `TeamLoader`.  The `LoaderPackage` is to interface with FTL and the TeamLoader.